music_assistant/server/controllers/music.py

```diff
@@ -157,6 +160,33 @@ async def search(
         :param media_types: A list of media_types to include.
         :param limit: number of items to return in the search (per type).
         """
+        # Check if the search query is a streaming provider public shareable URL
+        try:
+            media_type, provider_instance_id_or_domain, item_id = parse_uri(search_query)
```

Comment:
I think this helper util alone should be enough to parse a shareable URL. Calling get_item is heavy and overkill. We should improve the "parse_uri" if that doesn't already handle it. I know I added some support in there for share URL's.

Reply:
I am using `parse_uri` to parse the url and get the media_type and the item_id from it.  I then use  these in `get_item` to actually retrieve the `MediaType` item from the provider, then stuff the item in a `SearchResults` for return.  So there's no actual searching going on here, I'm just using the search interface to present the "results".

Reply:
ah yeah right... So instead of doing a text search on name, you can just paste an uri or share link.

Reply:
Still, if you know for certain that the spotify id's should fit a common pattern, I think it would be good to improve the parse_uri helper to catch this as early as possible and save the overhead of the URL call. You could even raise the InvalidUri exception

Reply:
Good idea.  I've added validation for the spotify ID and others can be added later once URL search is enabled and provider ID format known.